engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -1,4 +1,10 @@
 module InstanceVerification
+  def self.update_cache(remote_ip, system_login, product_id)
+    cache_key = [remote_ip, system_login, product_id].join('-')
+    # caches verification result to be used by zypper auth plugin
+    Rails.cache.write(cache_key, true, expires_in: 20.minutes)
```

Comment:
Is there a reason behind having this cache for 20 minutes and not for 24h for example?

Reply:
It was one hour before (2 years ago) and it's been 20 min for that long.
Although, that change would be out of scope for this PR, it would need its own issue/PR

Reply:
The 20 minutes is a more or less arbitrary value. We expect that one run of `zypper` completes in this time meaning we do the system lookup only once. Also it reduces the window where the credentials and metadata can be injected into another system.

Reply:
Please be aware that the cache expiration is not respected when using the default file system cache, and if we want it changed, it needs to be changed in https://github.com/SUSE/rmt/blob/master/engines/registration_sharing/package/rmt-server-trim-cache.service as well (as that's the only function that _actually_ expires those cache items).